From f2d6a175c6ece8fc38e633f4d4bde4da6bbd2d40 Mon Sep 17 00:00:00 2001
From: Tang Yuantian <Yuantian.Tang@nxp.com>
Date: Mon, 1 Feb 2016 14:47:38 +0800
Subject: [PATCH 0545/1383] cpufreq: enable 64bit soc support

Added ARM-based 64bit soc support for QorIQ platforms.
Also added two 64bit platform compatible string to the list
which are ls1043a and ls2080a.

Signed-off-by: Tang Yuantian <yuantian.tang@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/cpufreq/Kconfig         | 2 +-
 drivers/cpufreq/qoriq-cpufreq.c | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/cpufreq/Kconfig b/drivers/cpufreq/Kconfig
index 659879a..32c69e9 100644
--- a/drivers/cpufreq/Kconfig
+++ b/drivers/cpufreq/Kconfig
@@ -295,7 +295,7 @@ endif
 
 config QORIQ_CPUFREQ
 	tristate "CPU frequency scaling driver for Freescale QorIQ SoCs"
-	depends on OF && COMMON_CLK && (PPC_E500MC || ARM)
+	depends on OF && COMMON_CLK && (PPC_E500MC || ARM || ARM64)
 	select CLK_QORIQ
 	help
 	  This adds the CPUFreq driver support for Freescale QorIQ SoCs
diff --git a/drivers/cpufreq/qoriq-cpufreq.c b/drivers/cpufreq/qoriq-cpufreq.c
index 039ed85..d42caa5 100644
--- a/drivers/cpufreq/qoriq-cpufreq.c
+++ b/drivers/cpufreq/qoriq-cpufreq.c
@@ -287,6 +287,8 @@ static const struct of_device_id node_matches[] __initconst = {
 	{ .compatible = "fsl,p4080-clockgen", },
 	{ .compatible = "fsl,qoriq-clockgen-1.0", },
 	{ .compatible = "fsl,qoriq-clockgen-2.0", },
+	{ .compatible = "fsl,ls2080a-clockgen", },
+	{ .compatible = "fsl,ls1043a-clockgen", },
 	{}
 };
 
-- 
2.8.1

